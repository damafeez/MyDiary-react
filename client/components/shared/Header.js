import React from 'react';
import { Link } from 'react-router-dom';
import './Header.scss';
import john from '../../assets/john.jpg';

export default () => (
  <header>
    <h1><Link to="/">MyDiary</Link></h1>
    <Link to="/profile"><img className="bg" src={john} alt="" /></Link>
  </header>
);
