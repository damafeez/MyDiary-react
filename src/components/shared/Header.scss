header {
  justify-content: space-between;
  top: 0;
}
header h1 {
  font-weight: 500;
}
header h1 a {
  color: var(--secondary-color)
}
header img.bg {
  width: 3rem;
  height: 3rem;
  border-radius: 50%;
  box-shadow: 0 16px 26px -10px rgba(60, 72, 88, 0.56),
  0 4px 25px 0px rgba(0, 0, 0, 0.12),
  0 8px 10px -5px rgba(60, 72, 88, 0.2);
  cursor: pointer;
}