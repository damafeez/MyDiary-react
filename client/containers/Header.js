import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import './Header.scss';
import { userPlaceholderImage } from '../mixin';

export const Header = ({ profileImage }) => (
  <header>
    <h1><Link to="/">MyDiary</Link></h1>
    <Link to="/profile"><img className="bg" src={profileImage || userPlaceholderImage} alt="" /></Link>
  </header>
);

export default connect((state) => ({ profileImage: state.auth.user.image }), {})(Header);
