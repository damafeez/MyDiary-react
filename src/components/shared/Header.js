import React from 'react';
import { Link } from 'react-router-dom';
import './Header.scss';
import john from '../../assets/john.jpg';

export default (props) => {
  const {
    type, label, handleChange, value, name, required, icon,
  } = props;
  return (
    <header>
      <h1><a href="home.html">MyDiary</a></h1>
      <Link to="/profile"><img className="bg" src={john} alt="" /></Link>
    </header>
  );
};
