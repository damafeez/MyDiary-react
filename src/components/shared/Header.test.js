import React from 'react';
import Header from './Header';

it('matches snapshot', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});
