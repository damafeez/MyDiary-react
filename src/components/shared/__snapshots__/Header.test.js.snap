// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          <Link
            replace={false}
            to="/"
          >
            MyDiary
          </Link>
        </h1>,
        <Link
          replace={false}
          to="/profile"
        >
          <img
            alt=""
            className="bg"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/"
          >
            MyDiary
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "MyDiary",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "MyDiary",
          "type": [Function],
        },
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            alt=""
            className="bg"
            src={
              Object {
                "process": [Function],
              }
            }
          />,
          "replace": false,
          "to": "/profile",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "bg",
            "src": Object {
              "process": [Function],
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            <Link
              replace={false}
              to="/"
            >
              MyDiary
            </Link>
          </h1>,
          <Link
            replace={false}
            to="/profile"
          >
            <img
              alt=""
              className="bg"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              MyDiary
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "MyDiary",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "MyDiary",
            "type": [Function],
          },
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt=""
              className="bg"
              src={
                Object {
                  "process": [Function],
                }
              }
            />,
            "replace": false,
            "to": "/profile",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "bg",
              "src": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
